src/Runner.Worker/ExecutionContext.cs

```diff
@@ -1043,7 +1045,7 @@ private IExecutionContext CreatePostChild(string displayName, Dictionary<string,
             }
 
             var newGuid = Guid.NewGuid();
-            return CreateChild(newGuid, displayName, newGuid.ToString("N"), null, null, ActionRunStage.Post, intraActionState, _childTimelineRecordOrder - Root.PostJobSteps.Count, siblingScopeName: siblingScopeName);
```

Comment:
I'm not sure this covers all the edge cases. The problem in question is this [line](https://github.com/actions/runner/blob/ec9830836b214bb7564277c3d2b53e1303e191ae/src/Runner.Worker/ExecutionContext.cs#L351). In particular, the call from [CreateEmbeddedChild](https://github.com/actions/runner/blob/ec9830836b214bb7564277c3d2b53e1303e191ae/src/Runner.Worker/ExecutionContext.cs#L375)

Emedded Children shouldn't have an timeline order that matters, I think we can just define a constant (like 99999), and pass it into the [Createchild recordOrder](https://github.com/actions/runner/blob/ec9830836b214bb7564277c3d2b53e1303e191ae/src/Runner.Worker/ExecutionContext.cs#L375) call here, that way we enter this [side](https://github.com/actions/runner/blob/ec9830836b214bb7564277c3d2b53e1303e191ae/src/Runner.Worker/ExecutionContext.cs#L347) of this if condition, rather then the side that iterates `_childTimelineRecordOrder`

Reply:
While tracking the execution, Children would be created first, and then, when you hit [line](https://github.com/actions/runner/blob/15061a558c6d86a2b0da509f23b2e8e97b1d80f1/src/Runner.Worker/ActionRunner.cs#L139), it will attempt to call RegisterPostJobStep, which calls CreatePostChild, where the change was made. The way it worked is that you would set the timeline as  `2*NUMBER_OF_STEPS - NUMBER_OF_POST_STEPS`, which is a read-only property. That way, every POST_STEP would have the same timeline record and that is where the issue was found. By doing this, we are always setting the timeline to `2*NUMBER_OF_STEPS` and then, since the post steps should be in reverse order, and are called in the reverse order, we would just have one counter which would create a timeline record from `2*NUMBER_OF_STEPS-MAIN_STEP_INDEX`. For example, if we have:

1. checkout (has post action)
2. custom_action (does not have post action)
3. setup-node (has post action)

the timeline would be:
job: 1, checkout step: 2, custom_step: 3, setup_node: 4, checkout_post: 8, setup_node_post: 7

I did not want to change the way the logic works a lot, but since the previous code was a constant `(2 * NUMBER_OF_STEPS - NUMBER_OF_POST_STEPS) `, checkout_post and setup_node_post would both get the 8 and both hit the execution path you have described with [Createchild recordOrder](https://github.com/actions/runner/blob/ec9830836b214bb7564277c3d2b53e1303e191ae/src/Runner.Worker/ExecutionContext.cs#L375). This way, we are still hitting that execution path, but we are setting timeline attribute in reverse order from `2*NUMBER_OF_STEPS` to `NUMBER_OF_STEPS`